opensrs/opensrsapi.py

```diff
@@ -405,6 +411,8 @@ def register_domain(self, domain, purchase_period, user, user_id,
                             'Invalid syntax on domain')):
                     raise errors.InvalidDomain(e)
                 raise errors.DomainRegistrationFailure(e)
+            if e.response_code == self.CODE_CLIENT_TIMED_OUT:
+                raise errors.DomainRegistrationTimedOut(e)
```

Comment:
I can add that logging to domainservice as part of this work.
I think it might be easier to log first then add the exceptions we need.  OpenSRS reuses error codes for a lot of different errors. There are also some that aren't documented.

I don't think things like handling `SSLError`s should be done in the OpenSRS library though.